contexts/WalletManagerContext.tsx

```diff
@@ -649,6 +647,10 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
     }
 
     const fetchBalance = async () => {
+      if (isWalletSettingsLoading) {
+        return;
+      }
+
```

Comment:
I would move this check to the top of `useEffect`, say:
```
644    if (!rpcClient | isWalletSettingsLoading) {
      　　return;
   　　}
```

And add `isWalletSettingsLoading` to the dependency of useEffect.

Reply:
c28d67b0f1240fbc0bdc8a403eedada847e5ee73

Reply:
I move the check to the top of `useEffect`:
https://github.com/forbole/kastle/pull/21/commits/afc434afe0a7ac3661c1c1e74dd440cb5a069ecd